tests/testthat/test-template.R

```diff
@@ -29,21 +29,29 @@ test_that("Code blocks are evaluated and rendered correctly", {
 })
 
 test_that("UTF-8 characters in templates", {
-  template <- htmlTemplate("template-document.html", x = "")
-  html <- renderDocument(template)
-
-  # Create the string 'Δ★😎', making sure it's UTF-8 encoded on all platforms.
-  # These characters are 2, 3, and 4 bytes long, respectively.
-  pat <- rawToChar(as.raw(c(0xce, 0x94, 0xe2, 0x98, 0x85, 0xf0, 0x9f, 0x98, 0x8e)))
-  Encoding(pat) <- "UTF-8"
-  expect_true(grepl(pat, html))
-
-  # If template is passed text_ argument, make sure it's converted from native
-  # to UTF-8.
-  latin1_str <- rawToChar(as.raw(0xFF))
-  Encoding(latin1_str) <- "latin1"
-  text <- as.character(htmlTemplate(text_ = latin1_str))
-  expect_identical(charToRaw(text), as.raw(c(0xc3, 0xbf)))
+  test_template <- function(locale=""){
```

Comment:
The implementation here is unchanged -- just adding the ability to set the locale prior to running it. When using the "alternate" `paste8` implementation proposed in 01fb9, this test fails only when running in the Chinese locale.